cmd/reporting-operator/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jackc/pgx/v4"
+	"github.com/operate-first/curator-operator/internal"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/cobra"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	defaultServerPort = ":8080"
+	defaultTimeout    = time.Second * 10
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+func newCmd() *cobra.Command {
+	var (
+		serverPort string
+	)
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts the http server",
+		Long:  "",
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
blocking: Use the [`RunE`](https://pkg.go.dev/github.com/spf13/cobra#Command) function as it has an error return type:

> Note: change the runReporting to runServer. See an implementation below.

```go
if err := runServer(context.Background(), reportLog, serverPort); err != nil {
    return err
}
```